tests/test_ERC721_Pausable.py

```diff
@@ -0,0 +1,180 @@
+
+
+@pytest.fixture(scope='module')
+async def erc721_factory():
+    starknet = await Starknet.empty()
+    owner = await starknet.deploy(
+        "contracts/Account.cairo",
+        constructor_calldata=[signer.public_key]
+    )
+
+    other = await starknet.deploy(
+        "contracts/Account.cairo",
+        constructor_calldata=[signer.public_key]
+    )
+
+    erc721 = await starknet.deploy(
+        "contracts/token/ERC721_Pausable.cairo",
+        constructor_calldata=[
+            str_to_felt("Non Fungible Token"),  # name
+            str_to_felt("NFT"),                 # ticker
+            owner.contract_address              # owner
+        ]
+    )
+
+    erc721_holder = await starknet.deploy("contracts/token/utils/ERC721_Holder.cairo")
+
+    return starknet, erc721, owner, other, erc721_holder
+
+
+@pytest.mark.asyncio
+async def test_constructor(erc721_factory):
+    _, erc721, _, _, _ = erc721_factory
+    execution_info = await erc721.name().call()
+    assert execution_info.result == (str_to_felt("Non Fungible Token"),)
+
+    execution_info = await erc721.symbol().call()
+    assert execution_info.result == (str_to_felt("NFT"),)
+
+
+@pytest.mark.asyncio
+async def test_pause(erc721_factory):
+    _, erc721, owner, other, erc721_holder = erc721_factory
+
+    # mint tokens to account
+    tokens = [first_token_id, second_token_id]
+    for token in tokens:
+        await signer.send_transaction(
+            owner, erc721.contract_address, 'mint', [
+                owner.contract_address, *token]
+        )
+
+    # pause
+    await signer.send_transaction(owner, erc721.contract_address, 'pause', [])
+
+    execution_info = await erc721.paused().call()
+    assert execution_info.result.paused == 1
+
+    await assert_revert(signer.send_transaction(
+        owner, erc721.contract_address, 'approve', [
+            other.contract_address,
+            *first_token_id
+        ])
+    )
+
+    await assert_revert(signer.send_transaction(
+        owner, erc721.contract_address, 'setApprovalForAll', [
+            other.contract_address,
+            true
+        ])
+    )
+
+    await assert_revert(signer.send_transaction(
+        owner, erc721.contract_address, 'transferFrom', [
+            owner.contract_address,
+            other.contract_address,
+            *first_token_id
+        ])
+    )
+
+    await assert_revert(signer.send_transaction(
+        owner, erc721.contract_address, 'safeTransferFrom', [
+            owner.contract_address,
+            erc721_holder.contract_address,
+            *first_token_id,
+            len(data),
+            *data
+        ])
+    )
+
+    await assert_revert(signer.send_transaction(
+        owner, erc721.contract_address, 'mint', [
+            other.contract_address,
+            *third_token_id
+        ])
+    )
+
+
+@pytest.mark.asyncio
+async def test_unpause(erc721_factory):
+    _, erc721, owner, other, erc721_holder = erc721_factory
+
```

Comment:
maybe testing that it's paused? just to ensure a proper state before testing

Reply:
we really should refactor our test suites to make each case independant

Reply:
Excellent idea! I made the test cases independent here by changing the 721 factory scope to `function`. The one downside, however, consists of how long the tests take. I was also tempted to add `conftest.py` to keep account and contract fixtures, but I think it'd be more appropriate when we focus on reorganizing the repo

Reply:
we can have a memoized function to reduce cost